### DISCUSSION TITLE
Live tracking k8s resource by kubedog

### DISCUSSION THREAD

**Discussion by yxxhero** [helmfile/helmfile#660] [Ideas]
kubedog Github: https://github.com/werf/kubedog

screenshot:
![image](https://user-images.githubusercontent.com/11087727/215263754-a6c2a3c6-3def-458d-a9d6-0c2d12bcf2b9.png)

we can not only track k8s resources but also k8s logs.

**Comment by yxxhero** [comment]
## current track mode
helm --wait: it is very simple, can't get a log when workload has errors.

## new track mode
kubedog: it supports more format track style. it's very clear. detailed in the screenshot. and it can get workload logs by regex. it's very flexible.

**Reply by yxxhero** [reply]
so we can let users to choose the way to track.

**Comment by xiaomudk** [comment]
My thoughts adding kubedog would bloat the helmfile. Helm --wait can not meet the scene, we can raise an issue to helm.

**Comment by yxxhero** [comment]
I found kubedog in [helmware](https://github.com/helmwave/helmwave), a lot of features of helmware come from helmfile. contains config file format. there is a special feature that kubedog.

**Comment by yxxhero** [comment]
I will work on this feature.

**Comment by felipecrs** [comment]
OMG this will be an awesome feature, especially useful in CI.

**Reply by yxxhero** [reply]
@felipecrs are you interested in this feature?

**Reply by felipecrs** [reply]
Yes. You won't believe what I did to circumvent ~somewhat:

1. I put helmfile and stern inside a docker image
2. I start a script in background of the image entrypoint
3. I have helmfile hooks for presync and postsync that signals to my script in the background that it should start/stop stern.
4. I also have kubectl get events running in watch mode when I trigger this script.

But it is full of problems.

**Reply by felipecrs** [reply]
On the other hand, I don't know how this could be done with the current way how helmfile works... I mean, it delegates to the helm CLI, so it loses the track of the underlying k8s resources being changed.

I suppose this is one of the reasons why helmwave does not use helm CLI.

**Reply by yxxhero** [reply]
@felipecrs seems like this feature is valuable.

**Reply by felipecrs** [reply]
It definitely is, and many people believe it should be a feature of helm CLI itself.

But it seems that helm CLI is not going anywhere... so we can't expect them to accept or implement such a thing.

https://github.com/helm/helm/issues/3481

**Reply by jouve** [reply]
same, I often run helmfile+stern+`kubectl get pod -w` side by side in tmux

**Comment by jouve** [comment]
@xiaomudk  helm seems "stuck", for example, waiting for https://github.com/helm/helm/pull/9950 for almost 2 years

**Reply by shcherbak** [reply]
no updates?
